Sources/StreamChatUI/MessageActionsPopup/ChatMessagePopupVC.swift

```diff
@@ -25,24 +35,36 @@ open class _ChatMessagePopupVC<ExtraData: ExtraDataTypes>: _ViewController, UICo
             .withoutAutoresizingMaskConstraints
     }()
 
-    public private(set) lazy var messageContentView = uiConfig.messageList.messageContentView.init()
+    /// New instance of `messageContentViewClass` that is populated with `message` data
+    open private(set) lazy var messageContentView = messageContentViewClass.init()
         .withoutAutoresizingMaskConstraints
 
+    /// `messageContentView` class that is populated with `message` and shown
+    public var messageContentViewClass: _ChatMessageContentView<ExtraData>.Type!
+    /// Message data that is shown
     public var message: _ChatMessageGroupPart<ExtraData>!
+    /// Initial frame of a message
     public var messageViewFrame: CGRect!
-    public var originalMessageView: UIView!
+    /// `_ChatMessageActionsVC` instance for showing actions
     public var actionsController: _ChatMessageActionsVC<ExtraData>!
+    /// `_ChatMessageReactionsVC` instance for showing reactions.
     public var reactionsController: _ChatMessageReactionsVC<ExtraData>?
-
-    // MARK: - Private
-
-    private var actionsView: UIView { actionsController.view }
-    private var actionsViewHeight: CGFloat { CGFloat(actionsController.messageActions.count) * 40 }
-    private var reactionsView: UIView? { reactionsController?.view }
-    private var reactionsViewHeight: CGFloat { reactionsView == nil ? 0 : 40 }
-    private var spacing: CGFloat = 8
-
-    // MARK: - Life Cycle
+    
+    /// Spacing between reactions, message, and actions
+    public var spacing: CGFloat = 8
+    
+    /// Layout properties of this view
+    public private(set) var layout = Layout()
+    
+    /// Properties tied to `_ChatMessagePopupVC` layout
+    public struct Layout {
+        /// Constraints of `reactionsController.view`
+        public fileprivate(set) var reactionsViewConstraints: [NSLayoutConstraint] = []
+        /// Constraints of `actionsController.view`
+        public fileprivate(set) var actionsViewConstraints: [NSLayoutConstraint] = []
+        /// Constraints of `messageContentView`
+        public fileprivate(set) var messageContentViewConstraints: [NSLayoutConstraint] = []
+    }
```

Comment:
Ideally, we should no longer need to do this 🤔 Can't we just put everything into a `ContainerView` and lay it out using that?

Reply:
true, deleted